Onboarding Newsletter toggle and user interview request

jpaddison3 opened the pull request:
<img width="759" alt="image" src="https://github.com/user-attachments/assets/9cfb2267-33c5-4219-89d6-257fb57a7153" />

I've commented out the podcast code, as we might add it back in later.

I could see other choices being made wrt refactoring the mailchimp functionality, given that we now have three lists, but chose to leave them as three separate fields, but combine the digest and newsletter update functions.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1209995250335420) by [Unito](https://www.unito.io)

jpaddison3 commented:
@agnesstenlund

Will-Howard commented on packages/lesswrong/server/callbacks/userCallbackFunctions.tsx:
```diff
@@ -329,19 +329,24 @@ export async function changeDisplayNameRateLimit(validationErrors: CallbackValid
   if (hasDigests && (unsubscribedFromAll || deactivatedAccount)) {
     data.subscribedToDigest = false
   }
+  if (hasNewsletter && (unsubscribedFromAll || deactivatedAccount)) {
+    data.subscribedToNewsletter = false
+  }
```
I don't think this should unsubscribe you from the newsletter (I'm pretty on the fence about the digest even, but the newsletter is even more separate from the forum)

jpaddison3 replied:
I agree if they subscribed to the newsletter from somewhere else, but not if they subscribed from the Forum, which they would have done here. I'm not particularly on the fence about the digest, personally.

Will-Howard commented on packages/lesswrong/server/callbacks/userCallbackFunctions.tsx:
```diff
@@ -352,59 +357,73 @@ export async function updateDigestSubscription(data: Partial<DbUser>, {oldDocume
+  const updates = [
+    {noUpdate: noDigestUpdate, listId: mailchimpForumDigestListId, status: digestStatus},
+    {noUpdate: noNewsletterUpdate, listId: mailchimpEANewsletterListId, status: newsletterStatus}
+  ].filter((u) => !!u.listId)
```
Could you just filter the noUpdate ones out here so they don't have to be skipped again below

jpaddison3 replied:
Obviously 🤦

Will-Howard commented on packages/lesswrong/server/callbacks/userCallbackFunctions.tsx:
```diff
@@ -352,59 +357,73 @@ export async function updateDigestSubscription(data: Partial<DbUser>, {oldDocume
+  const mailchimpEANewsletterListId = mailchimpEAForumNewsletterListIdSetting.get();
+  const digestUpdateButNoListId = hasDigests && !mailchimpForumDigestListId
+  const newsletterUpdateButNoListId = hasNewsletter && !mailchimpEANewsletterListId
+  if (!mailchimpAPIKey || digestUpdateButNoListId || newsletterUpdateButNoListId) {
```
```suggestion
  if (!mailchimpAPIKey || (digestUpdateButNoListId && newsletterUpdateButNoListId)) {
```
It should just skip if one is missing

jpaddison3 replied:
Kinda. It should still record an error.

Will-Howard reviewed (approved):
ASTLAC, I tested that it creates the subscription as expected

jpaddison3 commented:
@agnesstenlund is this what you mean?
<img width="559" alt="image" src="https://github.com/user-attachments/assets/2b87e4f5-d4db-4ba6-a7e8-e3eadb9b8f31" />

agnesstenlund commented:
Sorry no I should've specified better 🙈 meant the blue button below is right-aligned with the content, but doesn't go all the way to the left 
![image](https://github.com/user-attachments/assets/3ef8b5ca-3237-42ba-9698-d3fad95aaa7e)

jpaddison3 commented:
Note to self: lemme double check this on mobile.

jpaddison3 commented:
@agnesstenlund the answer to the right-side-strip first.

The strip is an empty container for a scrollbar. We configure it to be there. I'm not sure why, and it's there on master. I've removed it, but @oetherington could tell me if he remembers why we set that `overflow-y: 'scroll'`.

Which the leads to the answer to the button misalignment. The button looks a lot better once the scrollbar is removed:

<img width="628" alt="image" src="https://github.com/user-attachments/assets/32e73ef6-0048-4ca4-bc5c-e25c5f1b3b96" />

However, it doesn't match the original designs, nor the way you thought it should be this time. I've corrected it by adding some padding, but @s-cheng might remember if there was a reason that didn't work when she last touched the button width. New version:

<img width="627" alt="image" src="https://github.com/user-attachments/assets/b5905119-faab-4431-87cd-2a8c61b68897" />

(To my mind both look about equally good, and I'd be happy to go either way.)

jpaddison3 commented:
I'm gonna go ahead and merge, but happy to make changes.
